src/commands/dev/index.js

```diff
@@ -35,47 +34,59 @@ const startStaticServer = async ({ settings, log }) => {
   log(`\n${NETLIFYDEVLOG} Server listening to`, settings.frameworkPort)
 }
 
+const NOT_FOUND_EXIT_CODE = 127
+
+const isNonExistingCommandError = (error) => {
+  // `ENOENT` is only returned for non Windows systems
+  // See https://github.com/sindresorhus/execa/pull/447
+  // exitCode === 127 is returned from npm/yarn
+  if (error.code === 'ENOENT' || error.exitCode === NOT_FOUND_EXIT_CODE) {
+    return true
+  }
+
+  // this only works on English versions of Windows
+  return (
+    typeof error.message === 'string' && error.message.includes('is not recognized as an internal or external command')
+  )
+}
+
 const startFrameworkServer = async function ({ settings, log, exit }) {
   if (settings.noCmd) {
     return await startStaticServer({ settings, log })
   }
 
   log(`${NETLIFYDEVLOG} Starting Netlify Dev with ${settings.framework || 'custom config'}`)
-  const commandBin = await which(settings.command).catch((error) => {
-    if (error.code === 'ENOENT') {
-      throw new Error(
-        `"${settings.command}" could not be found in your PATH. Please make sure that "${settings.command}" is installed and available in your PATH`,
+
+  const commandWithArgs = `${settings.command} ${settings.args.join(' ')}`
+  // we use reject=false to avoid rejecting synchronously when the command doesn't exist
+  // we can't try->await->catch since we don't want to block on the framework server which
+  // is a long running process
+  const frameworkProcess = execa.command(commandWithArgs, { preferLocal: true, reject: false })
+  frameworkProcess.stdout.pipe(stripAnsiCc.stream()).pipe(process.stdout)
+  frameworkProcess.stderr.pipe(stripAnsiCc.stream()).pipe(process.stderr)
+  process.stdin.pipe(frameworkProcess.stdin)
+
+  // eslint-disable-next-line promise/catch-or-return,promise/prefer-await-to-then
+  frameworkProcess.then(async () => {
```

Comment:
Should we use `await`?

Reply:
`frameworkProcess` is for example the invocation of `gatsby develop`. We can't `await` it since it will block the CLI.
We still want to know when it exists since we should exit the CLI with an error when that happens.

Reply:
Makes sense!
We might want to add a code comment for future reference.

Reply:
It's commented here https://github.com/netlify/cli/pull/2857/files#diff-42c491dfbe209d7493ec82cd5d031be64f17cf6e622201768d29b3539194d9e1R61

I'll move part of that comment closer to the `then`